guides/source/upgrading_ruby_on_rails.md

```diff
@@ -26,6 +26,38 @@ Rails generally stays close to the latest released Ruby version when it's releas
 
 It's a good idea to upgrade Ruby and Rails separately. Upgrade to the latest Ruby you can first, and then upgrade Rails.
 
+#### Ruby 3 breaking changes
+
+Ruby 3 introduces breaking changes to [keyword arguments](https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/). Whilst we do our best in order to make sure the Rails API is compatible between supported Ruby versions sometimes this isn't possible, below you'll find some (but not all) examples of such issues.
+
+Passing the keyword argument as the last hash parameter is no longer working, you need to transform the hash parameter to keywords by using the double splat operator `**`:
```

Comment:
Some minor suggestions to simplify the wording slightly. Please take this with a grain of salt as this is mostly a matter of personal preference. 🤗 

```suggestion
Ruby 3 introduces breaking changes to [keyword arguments](https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/). While we do our best in order to make sure the Rails API is compatible between supported Ruby versions sometimes this isn't possible. Below you'll find an example of such an issue.

Passing the keyword argument as the last hash parameter is no longer allowed. You need to use keywords or transform the hash parameter to keywords by using the double splat operator `**`:
```

Reply:
I think the "in order" can be removed...
"While we do our best to make sure the Rails API is compatible..."